tests/services/leaderboard.test.js

```diff
@@ -0,0 +1,40 @@
+import { fetchLeaderboardToppers } from "@services/apiService";
+
+const API_URL = process.env.NEXT_PUBLIC_API_LINK;
+global.fetch = jest.fn();
+
+describe("fetchLeaderboardToppers", () => {
+  afterEach(() => {
+    fetch.mockClear();
+  });
+
+  it("fetches leaderboard toppers correctly", async () => {
+    const mockResponse = { leaderboardData: "mockedData" };
+    const params = { addr: "exampleAddr", duration: "exampleDuration" };
+
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockResponse),
+    });
+
+    const response = await fetchLeaderboardToppers(params);
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/leaderboard/get_static_info?addr=${params.addr}&duration=${params.duration}`
+    );
+    expect(response).toEqual(mockResponse);
+  });
+
+  it("handles errors gracefully", async () => {
+    const params = { addr: "exampleAddr", duration: "exampleDuration" };
+
+    const mockResponse = "Leaderboard positions not found";
```

Comment:
is this the response you get when there is no position found?
i think it should be something like - 'Error querying ranks'

Reply:
Should I edit the response?

Reply:
i just modified the response. opening a new pull request

Reply:
> is this the response you get when there is no position found? i think it should be something like - 'Error querying ranks'
I’ve edited the response message